Skyline.DataMiner.CICD.Tools.Validator/ValidatorRunner.cs

```diff
@@ -0,0 +1,230 @@
+            }
+
+            if (outputFormats.Any(f => String.Equals(f, "JSON", StringComparison.OrdinalIgnoreCase)))
+            {
+                resultWriters.Add(new ResultWriterJson(Path.Combine(validatorResultsOutputDirectory, $"{validatorResultsFileName}.json"), logger));
+            }
+
+            if (outputFormats.Any(f => String.Equals(f, "HTML", StringComparison.OrdinalIgnoreCase)))
+            {
+                resultWriters.Add(new ResultWriterHtml(Path.Combine(validatorResultsOutputDirectory, $"{validatorResultsFileName}.html"), logger));
+            }
+
+            await SendMetricAsync("protocol", "solution");
+
+            logger.LogInformation("Writing results...");
+            foreach (var writer in resultWriters)
+            {
+                writer.WriteResults(validatorResults);
+            }
+
+            logger.LogInformation("Writing results completed");
+
+            logger.LogInformation("Finished");
+            return 0;
+        }
+
+        private static bool IsLegacyStyleSolution(Solution solution)
+        {
+            bool isLegacyStyleSolution = false;
+
+            foreach (var p in solution.Projects)
+            {
+                var project = solution.LoadProject(p);
+
+                if (project.ProjectStyle == ProjectStyle.Legacy)
+                {
+                    isLegacyStyleSolution = true;
+                    break;
+                }
+            }
+
+            return isLegacyStyleSolution;
+        }
+
+        private static bool IsTargetingNetFramework48(Solution solution)
+        {
+            bool isTargetingNetFramework48 = true;
+
+            foreach (var p in solution.Projects)
+            {
+                var project = solution.LoadProject(p);
+                if (!project.TargetFrameworkMoniker.StartsWith(".NETFramework,Version=v4.8"))
+                {
+                    isTargetingNetFramework48 = false;
+                    break;
+                }
+            }
+
+            return isTargetingNetFramework48;
+        }
+
+        private void BuildSolution(string solutionFilePath, int buildTimeout)
+        {
+            using (Process process = new Process())
+            {
+                process.StartInfo.FileName = "dotnet";
+                process.StartInfo.Arguments = $"build \"{solutionFilePath}\" /property:WarningLevel=0";
+                process.StartInfo.UseShellExecute = false;
+                process.StartInfo.CreateNoWindow = true;
+                process.StartInfo.RedirectStandardOutput = true;
+                process.StartInfo.RedirectStandardError = true;
+
+                process.OutputDataReceived += (sender, args) => { if (args.Data != null) logger.LogInformation(args.Data); };
+                process.ErrorDataReceived += (sender, args) => { if (args.Data != null) logger.LogError(args?.Data); };
```

Comment:
args?.Data can be changed to args.Data as it will never be null due to the if check (SonarCloud)